Application Form: limit FRQ answer length

The `Textfield` inputs currently do not have a max length, so an applicant could enter a quite lengthy answer which would be rejected by the API with status 422 which won't be surface during the client-side submission.

To avoid this, we can set a `maxLength` on the `textarea` fields which corresponds with the limit in the `ApplicationData` model. This is currently 1024 characters (~200 words), whereas the form label suggests 100/150 words, but I'm not sure if this is a soft or hard limit, so we may need to adjust the character limit accordingly.

Update, the API will actually throw status 405
> [ERROR] ValidationError: 1 validation error for RawApplicationData
> String should have at most 1024 characters [type=string_too_long, input_value="...", input_type=str]

We received a user report today of seeing status 405 in the browser console with no other indication.

--- a/apps/api/src/models/ApplicationData.py
+++ b/apps/api/src/models/ApplicationData.py
@@ -31,8 +31,8 @@ class RawApplicationData(BaseModel):
     is_first_hackathon: bool
     linkedin: Union[HttpUrl, None] = None
     portfolio: Union[HttpUrl, None] = None
-    frq_collaboration: Union[str, None] = Field(None, max_length=1024)
-    frq_dream_job: str = Field(max_length=1024)
+    frq_collaboration: Union[str, None] = Field(None, max_length=2048)
+    frq_dream_job: str = Field(max_length=2048)
 
 
 class ProcessedApplicationData(RawApplicationData):

--- a/apps/site/src/app/apply/sections/Components/Textfield.tsx
+++ b/apps/site/src/app/apply/sections/Components/Textfield.tsx
@@ -7,6 +7,7 @@ interface TextfieldProps {
 	inputClass: string;
 	containerClass: string;
 	isRequired: boolean;
+	maxLength?: number;
 }
 
 export default function Textfield({
@@ -16,6 +17,7 @@ export default function Textfield({
 	inputClass,
 	containerClass,
 	isRequired,
+	maxLength,
 }: TextfieldProps) {
 	return (
 		<div className={containerClass}>
@@ -29,6 +31,7 @@ export default function Textfield({
 					id={name}
 					name={name}
 					required={isRequired}
+					maxLength={maxLength}
 				/>
 			</div>
 		</div>

--- a/apps/site/src/app/apply/sections/Form/ProfileInformation.tsx
+++ b/apps/site/src/app/apply/sections/Form/ProfileInformation.tsx
@@ -2,6 +2,8 @@ import TextInput from "@/app/apply/sections/Components/TextInput";
 import Textfield from "@/app/apply/sections/Components/Textfield";
 import styles from "./Form.module.scss";
 
+const FRQ_MAX_LENGTH = 2000;
+
 export default function ProfileInformation() {
 	return (
 		<div className="flex flex-col gap-5 w-11/12">
@@ -39,6 +41,7 @@ export default function ProfileInformation() {
 				inputClass={`bg-[#E1E1E1] p-3 h-48 resize-none rounded-xl`}
 				containerClass="flex flex-col w-full"
 				isRequired={true}
+				maxLength={FRQ_MAX_LENGTH}
 			/>
 
 			<Textfield
@@ -48,6 +51,7 @@ export default function ProfileInformation() {
 				inputClass={`bg-[#E1E1E1] p-3 h-48 resize-none rounded-xl`}
 				containerClass="flex flex-col w-full"
 				isRequired={true}
+				maxLength={FRQ_MAX_LENGTH}
 			/>
 		</div>
 	);

